discord/embeds.py

```diff
@@ -383,15 +386,20 @@ def set_author(self, *, name, url=EmptyEmbed, icon_url=EmptyEmbed):
             The URL of the author icon. Only HTTP(S) is supported.
         """
 
-        self._author = {
-            'name': str(name)
-        }
+        if name is EmptyEmbed:
+            if url is not EmptyEmbed or icon_url is not EmptyEmbed:
+                raise ValueError("can't set author without name")
+            del self._author
+        else:
```

Comment:
I don't like over-indenting too much. I suggest returning from here and leaving the rest unindented.